specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-10-01-preview/machineLearningServices.json

```diff
@@ -5848,17 +5850,8 @@
       }
     },
     "ConnectionCategory": {
```

Comment:
>ConnectionCategory [](http://example.com/codeflow?start=5&length=18)

Is it not possible add categories, rather than change it to unconstrained string?

Reply:
we are supporting more and more categories and updating swagger each time won't scale well. so we plan to change this back to a string, but we will check on server side and return an error message with supported categories.

Reply:
Not sure I get it. How can you add categories without a contract change? Contract change will require an API version change. Once you do that, it's easy to add enum values. In fact, you can add enum values without an API version change as well, as long as it sets modelAsString in the x-ms-enum extension. I don't see the advantage of unconstrained string.

Reply:
A constrained string needs to be modeled as an enum. Its not a good practice to accept a string and return a bad request if its not in the set u expect it to be. How do u expect your customers to know what the possible values are?